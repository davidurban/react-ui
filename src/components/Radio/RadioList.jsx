import PropTypes from 'prop-types';
import React from 'react';
import UserInputBase from '../UserInputBase';

/* eslint-disable jsx-a11y/label-has-for */
const RadioList = (props) => {
  const {
    options,
    ...higherOrderProps
  } = props;
  return (
    <ul {...higherOrderProps}>
      {
        options.map(option => (
          <li key={option}>
            <label>
              <input
                disabled={higherOrderProps.disabled}
                name={higherOrderProps.id}
                type="radio"
              />
              { option }
            </label>
          </li>
        ))
      }
    </ul>
  );
};

RadioList.propTypes = {
  options: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default UserInputBase(RadioList);
