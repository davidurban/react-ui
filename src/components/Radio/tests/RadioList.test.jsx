import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import RadioList from '../index';

describe('rendering', () => {
  it('renders correctly mandatory props only', () => {
    const tree = shallow(<RadioList
      fieldId="test"
      options={['choice 1', 'choice2']}
    />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly only one option', () => {
    const tree = shallow(<RadioList
      fieldId="test"
      options={['choice 1']}
    />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly with all props', () => {
    const tree = shallow(<RadioList
      label="label"
      disabled
      fieldId="test"
      value="value"
      helpText="some help"
      errors={['some error', 'another error']}
      required
      options={['choice 1', 'choice2']}
    />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
