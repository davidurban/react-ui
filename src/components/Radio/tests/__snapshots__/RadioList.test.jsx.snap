// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly mandatory props only 1`] = `
<div>
  <RadioList
    disabled={false}
    id="test"
    options={
      Array [
        "choice 1",
        "choice2",
      ]
    }
    required={false}
  />
</div>
`;

exports[`rendering renders correctly only one option 1`] = `
<div>
  <RadioList
    disabled={false}
    id="test"
    options={
      Array [
        "choice 1",
      ]
    }
    required={false}
  />
</div>
`;

exports[`rendering renders correctly with all props 1`] = `
<div>
  <label
    htmlFor="test"
  >
    label
  </label>
  <RadioList
    disabled={true}
    id="test"
    options={
      Array [
        "choice 1",
        "choice2",
      ]
    }
    required={true}
    value="value"
  />
  <div>
    some help
  </div>
  <div
    key="some error"
  >
    some error
  </div>
  <div
    key="another error"
  >
    another error
  </div>
</div>
`;
